app/src/main/java/org/simple/clinic/bloodsugar/unitselection/BloodSugarUnitSelectionDialog.kt

```diff
@@ -132,6 +153,15 @@ class BloodSugarUnitSelectionDialog : AppCompatDialogFragment(), BloodSugarUnitS
     dismiss()
   }
 
+  override fun prefilledBloodSugarUnitSelection(blodSugarUnitPreference: BloodSugarUnitPreference) {
+    val bloodSugarUnitPreferenceButton: RadioButton? = when (blodSugarUnitPreference) {
+      BloodSugarUnitPreference.Mg -> bloodSugarUnitMg
+      BloodSugarUnitPreference.Mmol -> bloodSugarUnitMmol
+    }
+
+    bloodSugarUnitPreferenceButton?.isChecked = true
```

Comment:
Instead of this, there's a method on the `RadioGroup` called [`check()`](https://developer.android.com/reference/android/widget/RadioGroup#check(int)), which is the right method to use in this scenario since the radio buttons are wrapped in a radio group.